src/explore-education-statistics-common/src/modules/find-statistics/components/charts/AbstractChart.tsx

```diff
@@ -0,0 +1,126 @@
+import { CharacteristicsData } from '@common/services/tableBuilderService';
+import { Axis, ReferenceLine } from '@common/services/publicationService';
+import { Component, ReactNode } from 'react';
+import React from 'react';
+import * as Recharts from 'recharts';
+import { PositionType } from 'recharts';
+import { Label } from 'recharts';
+import { XAxisProps } from 'recharts';
+import { XAxis } from 'recharts';
+import { YAxisProps } from 'recharts';
+import { YAxis } from 'recharts';
+
+export interface ChartProps {
+  characteristicsData: CharacteristicsData;
+  chartDataKeys: string[];
+  labels: { [key: string]: string };
+  xAxis: Axis;
+  yAxis: Axis;
+  height?: number;
+  referenceLines?: ReferenceLine[];
+}
+
+export class AbstractChart<P extends ChartProps, S = {}> extends Component<
+  P,
+  S
+> {
+  protected calculateMargins() {
+    const margin = {
+      top: 15,
```

Comment:
Out of curiosity, is there any particularly way these margins are being determined? Or is is just trial and error?

I'm also wondering if these margins scale well responsively as well.

Reply:
At the moment it's trial and error. Recharts doesn't do responsive scaling for them, they are fixed to this size.